lgc/patch/PatchBufferOp.cpp

```diff
@@ -442,6 +442,33 @@ void PatchBufferOp::visitCallInst(CallInst &callInst) {
     m_replacementMap[&callInst] = std::make_pair(nullptr, nullptr);
 
     callInst.replaceAllUsesWith(numRecords);
+  } else if (callName.startswith(lgcName::LateBufferPtrDiff)) {
+    Value *const lhs = getPointerOperandAsInst(callInst.getArgOperand(0));
+    Value *const rhs = getPointerOperandAsInst(callInst.getArgOperand(1));
+
+    // If the pointer is not a fat pointer, bail.
+    Type *const lhsType = lhs->getType();
+    if (!lhsType->isPointerTy() || lhsType->getPointerAddressSpace() != ADDR_SPACE_BUFFER_FAT_POINTER)
+      return;
+
+    Type *const rhsType = rhs->getType();
+    if (!rhsType->isPointerTy() || rhsType->getPointerAddressSpace() != ADDR_SPACE_BUFFER_FAT_POINTER)
+      return;
```

Comment:
I think it would make sense to turn these checks into assertions.

Returning here would cause an invalid module anyway.